server/src/internalClusterTest/java/org/opensearch/search/fields/SearchIpFieldTermsIT.java

```diff
@@ -0,0 +1,249 @@
+            final String ip;
+            ip = generateRandomIPv4();
+            bulkRequestBuilder.add(
+                client().prepareIndex(indexName).setSource(Map.of("addr", ip, "dummy_filter", randomSubsetOf(1, "1", "2", "3")))
+            );
+            boolean match = false;
+            for (String termQ : queryTermsSink) {
+                boolean isCidr = termQ.contains("/");
+                if ((isCidr && isIPInCIDR(ip, termQ)) || (!isCidr && termQ.equals(ip))) {
+                    match = true;
+                    break;
+                }
+            }
+            if (match) {
+                addMatches++;
+            } else {
+                break; // single mismatch is enough.
+            }
+        }
+
+        bulkRequestBuilder.setRefreshPolicy(IMMEDIATE).get();
+        return ips + cidrs + addMatches;
+    }
+
+    public void testLessThanMaxClauses() throws IOException {
+        ArrayList<String> toQuery = new ArrayList<>();
+        String indexName = "small";
+        int expectMatches = createIndex(indexName, IndexSearcher.getMaxClauseCount() - 1, toQuery);
+
+        assertTermsHitCount(indexName, "addr", toQuery, expectMatches);
+        assertTermsHitCount(indexName, "addr.idx", toQuery, expectMatches);
+        assertTermsHitCount(indexName, "addr.dv", toQuery, expectMatches);
+        // passing dummy filter crushes on rewriting
+        try {
+            assertTermsHitCount(
+                indexName,
+                "addr.dv",
+                toQuery,
+                expectMatches,
+                (boolBuilder) -> boolBuilder.filter(QueryBuilders.termsQuery("dummy_filter", "1", "2", "3"))
+                    .filter(QueryBuilders.termsQuery("dummy_filter", "1", "2", "3", "4"))
+                    .filter(QueryBuilders.termsQuery("dummy_filter", "1", "2", "3", "4", "5"))
+            );
+            fail();
+        } catch (SearchPhaseExecutionException ose) {
+            assertTrue("exceeding on query rewrite", ose.shardFailures()[0].getCause() instanceof IndexSearcher.TooManyNestedClauses);
+        }
+    }
+
+    public void testExceedMaxClauses() throws IOException {
+        ArrayList<String> toQuery = new ArrayList<>();
+        String indexName = "larger";
+        int expectMatches = createIndex(indexName, IndexSearcher.getMaxClauseCount() + (rarely() ? 0 : atLeast(10)), toQuery);
+        assertTermsHitCount(indexName, "addr", toQuery, expectMatches);
+        assertTermsHitCount(indexName, "addr.idx", toQuery, expectMatches);
+        try { // error from mapper/parser
+            final SearchPhaseExecutionException ose = assertThrows(
+                SearchPhaseExecutionException.class,
+                () -> assertTermsHitCount(indexName, "addr.dv", toQuery, expectMatches)
+            );
+            assertTrue("exceeding on query building", ose.shardFailures()[0].getCause().getCause() instanceof IndexSearcher.TooManyClauses);
+            fail();
```

Comment:
The `fail()` and exception handling are not needed anymore: `assertThrows` takes care of that